#! /bin/sh /usr/share/dpatch/dpatch-run
## 10mono_cursor.dpatch
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: fix monochrome cursor issue

@DPATCH@

diff -u -r1.4.2.1 xrdp_constants.h
--- xrdp/common/xrdp_constants.h     1 May 2008 06:35:06 -0000       1.4.2.1
+++ xrdp/common/xrdp_constants.h     31 May 2008 02:12:08 -0000
@@ -213,6 +213,7 @@

 #define RDP_CAPSET_POINTER             8
 #define RDP_CAPLEN_POINTER             0x0a
+#define RDP_CAPLEN_POINTER_MONO        0x08

 #define RDP_CAPSET_SHARE               9
 #define RDP_CAPLEN_SHARE               0x08
diff -u -r1.9 rdp_rdp.c
--- xrdp/rdp/rdp_rdp.c       12 Jan 2007 04:51:48 -0000      1.9
+++ xrdp/rdp/rdp_rdp.c       31 May 2008 02:12:08 -0000
@@ -271,7 +271,7 @@
 rdp_rdp_out_pointer_caps(struct rdp_rdp* self, struct stream* s)
 {
   out_uint16_le(s, RDP_CAPSET_POINTER);
-  out_uint16_le(s, RDP_CAPLEN_POINTER);
+  out_uint16_le(s, RDP_CAPLEN_POINTER_MONO);
   out_uint16_le(s, 0); /* Color pointer */
   out_uint16_le(s, 20); /* Cache size */
   return 0;
@@ -355,7 +355,7 @@
   caplen = RDP_CAPLEN_GENERAL + RDP_CAPLEN_BITMAP + RDP_CAPLEN_ORDER +
            RDP_CAPLEN_BMPCACHE + RDP_CAPLEN_COLCACHE +
            RDP_CAPLEN_ACTIVATE + RDP_CAPLEN_CONTROL +
-           RDP_CAPLEN_POINTER + RDP_CAPLEN_SHARE +
+           RDP_CAPLEN_POINTER_MONO + RDP_CAPLEN_SHARE +
            0x58 + 0x08 + 0x08 + 0x34 /* unknown caps */  +
            4 /* w2k fix, why? */ ;
   if (rdp_sec_init(self->sec_layer, s, sec_flags) != 0)
